packages/plugin-constraints/sources/commands/constraints.ts

```diff
@@ -35,51 +35,62 @@ export default class ConstraintsCheckCommand extends BaseCommand {
 
   @Command.Path(`constraints`)
   async execute() {
-    for (let t = 0; t < 10; ++t) {
-      const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
-      const {project} = await Project.find(configuration, this.context.cwd);
-      const constraints = await Constraints.find(project);
+    const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
+    const {project} = await Project.find(configuration, this.context.cwd);
+    const constraints = await Constraints.find(project);
 
-      const toSave = new Set<Workspace>();
+    const report = await StreamReport.start({
+      configuration,
+      stdout: this.context.stdout,
+    }, async report => {
+      let allSaves = new Set<Workspace>();
+      const errors = new Set<[MessageName, string]>();
 
-      const report = await StreamReport.start({
-        configuration,
-        stdout: this.context.stdout,
-      }, async report => {
+      for (let t = 0, T = this.fix ? 10 : 1; t < T; ++t) {
+        errors.clear();
+
+        const toSave = new Set<Workspace>();
         const result = await constraints.process();
 
-        await processDependencyConstraints(toSave, result.enforcedDependencies, {
+        await processDependencyConstraints(toSave, errors, result.enforcedDependencies, {
           fix: this.fix,
           configuration,
-          report,
         });
 
-        await processFieldConstraints(toSave, result.enforcedFields, {
+        await processFieldConstraints(toSave, errors, result.enforcedFields, {
           fix: this.fix,
           configuration,
-          report,
         });
 
-        for (const workspace of toSave) {
-          workspace.persistManifest();
+        allSaves = new Set([
+          ...allSaves,
+          ...toSave,
+        ]);
+
+        // If we didn't apply any change, or there aren't any errors then we can exit the loop
+        if (toSave.size === 0 || errors.size === 0) {
```

Comment:
I think the second part is bogus - the fixable errors may appear after the save even if there isn't any unfixable error during the first run.
```suggestion
        // If we didn't apply any change then we can exit the loop
        if (toSave.size === 0) {
```

Reply:
fixable errors are not added to the errors Set and without that condition the fix will run al the 10 pass no matter what. 
Unless you believe that a fix to a fixable error might cause a new unfixable error I don't see why the loop should not break if we didn't find any error

Reply:
> without that condition the fix command will run all the 10 pass no matter what.

No, it will run an additional pass, by which point `toSave` will be empty, and it will bailout.

> Unless you believe that a fix to a fixable error might cause a new, unfixable error I don't see why the loop should not break if we didn't find any unfixable error.

Nope, I'm saying that a fix to a fixable error might cause a new other *fixable* error. For example, imagine there's a rule that require a dependency to a default range if it isn't set, and another that forces all dependencies to be prefixed with `foo:`. Solving the first one still requires to make a second pass to fix the second one.